src/clients/c/tb_client.zig

```diff
@@ -49,6 +53,77 @@ const TestingContext = blk: {
     break :blk ContextType(EchoClient);
 };
 
+/// Logging is global per application; it would be nice to be able to define a different logger for
+/// each client instance, though.
+var logging: Logging = .{};
+
+pub const Logging = struct {
+    const Callback = *const fn (
+        message_level: u8,
+        message_ptr: [*]const u8,
+        message_len: usize,
+    ) callconv(.C) void;
```

Comment:
Are we ok with the client library swallowing errors, or should this callback return `int` as an error code?

Reply:
zig's `defaultLog` swallows writer errors, so I think it's ok for us to assume the client library will do the same (or deal with it internally) too.